src/components/hooks/device-probe.ts

```diff
@@ -0,0 +1,117 @@
+import { useState, useEffect } from "react"
+import { IntlShape, useIntl } from "gatsby-plugin-intl"
+
+export type OS = "windows" | "mac" | "linux" | "ios" | "android"
+type Browser = "chrome" | "edge" | "ie" | "firefox" | "safari" | "opera" | "other"
+
+export type DeviceInfo = {
+  mobile: "yes" | "no" | "unknown"
+  os: OS | "unknown"
+  browser: Browser | "unknown"
+}
+
+function isKnownOS(os: string): os is OS {
```

Comment:
nitpicking but maybe we could have:
`[list,of,oses].includes(os)`

Reply:
Ah, nice